buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/dependencies/DependenciesMetadataRulesPlugin.kt

```diff
@@ -78,6 +78,10 @@ open class DependenciesMetadataRulesPlugin : Plugin<Project> {
                 // Test dependencies - minify: remove unused transitive dependencies
                 withLibraryDependencies("org.gradle.org.littleshoot:littleproxy", DependencyRemovalByNameRule::class,
                     setOf("barchart-udt-bundle", "guava", "commons-cli"))
+
+                // Gradle profiler should use the bundled tooling API
+                withLibraryDependencies("org.gradle.profiler:gradle-profiler", DependencyRemovalByNameRule::class,
+                    setOf("gradle-tooling-api"))
```

Comment:
Should we break up `gradle-profiler` in a way that we could depend on a tooling API-less variant/module?

Reply:
I don't think so. The right solution is that our project dependency should conflict with the external dependency and then the right dependency is selected. @jjohannes wanted to look into why this is not the case currently.

Reply:
This is the problem: https://github.com/gradle/gradle/issues/12002

I think this workaround here is ok for now. Maybe you can clarify this in the comment and add a link to the GH issue.

Reply:
I added a TODO.